tools/autograd/gen_variable_type.py

```diff
@@ -515,18 +515,19 @@ def find_args_with_derivatives(differentiable_inputs):
     if declaration['output_differentiability'] is not None:
         differentiable_outputs = []
         output_differentiability = declaration['output_differentiability']
+        if False in output_differentiability and inplace:
+            raise RuntimeError("output_differentiability=False for inplace operation (version_counter won't get updated)")
```

Comment:
Is it possible to check, for each argument, if it is being modified in place then output_differentiability should be True?

Reply:
Also please expand on the error message a little. Something like:
"we haven't implemented in-place behavior for output_differentiability=False outputs because it doesn't seem well defined"

Reply:
> Is it possible to check, for each argument, if it is being modified in place then output_differentiability should be True?
> 
> edit: If not, this is fine

I think we just have the single inplace boolean, not a per-argument one.

> Also please expand on the error message a little. Something like:
> "we haven't implemented in-place behavior for output_differentiability=False outputs because it doesn't seem well defined"

I feel like I want to be clear about why it's not well-defined or what would make it well defined in case someone needs to fix it. As someone who's not super familiar with output_differentiability, knowing that the version_counter is what will be wrong make it clearer to me. Is there a better explanation I could reference? I could update the error message as suggested and then also add a comment if it gets too verbose.

Reply:
> I think we just have the single inplace boolean, not a per-argument one.
Sounds good

> As someone who's not super familiar with output_differentiability, knowing that the version_counter is what will be wrong make it clearer to me.

That's fine

> I feel like I want to be clear about why it's not well-defined or what would make it well defined in case someone needs to fix it.

Yes, that's reasonable. 

> Is there a better explanation I could reference?

Not really. I don't understand `output_differentiability` 100%. `output_differentiability` implies that the output isn't differentiable. However, on an in-place operation, the output is the same as one of the inputs. If the input is differentiable, it's not clear to me if `output_differentiablity=True` means that the input suddenly becomes not-differentiable or if the output becomes different from the input.